File: frontend/src/types/evaluation.ts
export type EvaluationMode = 'baseline' | 'advanced';

export interface EvaluationMetric {
  key: string;
  label: string;
  unit: string;
  baseline: number;
  advanced: number;
  higherIsBetter: boolean;
}

export interface EvaluationCase {
  id: string;
  name: string;
  description: string;
  status: 'pending' | 'running' | 'completed' | 'failed';
  baselineScore: number;
  advancedScore: number;
  improvement: number;
  detailUrl?: string;
}

export interface EvaluationSummary {
  id: string;
  runAt: string;
  baselineOverall: number;
  advancedOverall: number;
  metrics: EvaluationMetric[];
  cases: EvaluationCase[];
}

export function mapEvaluationMetric(metric: {
  key: string;
  label: string;
  unit: string;
  baseline: number;
  advanced: number;
  higher_is_better: boolean;
}): EvaluationSummary['metrics'][number] {
  return {
    key: metric.key,
    label: metric.label,
    unit: metric.unit,
    baseline: metric.baseline,
    advanced: metric.advanced,
    higherIsBetter: metric.higher_is_better,
  };
}

export function mapEvaluationCase(item: {
  case_id: string;
  case_name: string;
  description: string;
  status: 'pending' | 'running' | 'completed' | 'failed' | string;
  baseline_score: number;
  advanced_score: number;
  improvement: number;
}): EvaluationSummary['cases'][number] {
  return {
    id: item.case_id,
    name: item.case_name,
    description: item.description,
    status: item.status as EvaluationSummary['cases'][number]['status'],
    baselineScore: item.baseline_score,
    advancedScore: item.advanced_score,
    improvement: item.improvement,
  };
}

export type EvaluationApiResponse = {
  id: string;
  status: string;
  benchmark_version: string;
  primary_metric: string;
  baseline_overall: number;
  advanced_overall: number;
  human_time_baseline: number;
  human_time_advanced: number;
  cost_baseline: number;
  cost_advanced: number;
  configuration?: Record<string, unknown> | null;
  created_at: string;
};

export type EvaluationDetailApiResponse = EvaluationApiResponse & {
  metrics: Array<{
    key: string;
    label: string;
    unit: string;
    baseline: number;
    advanced: number;
    higher_is_better: boolean;
  }>;
  cases: Array<{
    case_id: string;
    case_name: string;
    description: string;
    status: string;
    baseline_score: number;
    advanced_score: number;
    improvement: number;
  }>;
};

export function mapEvaluation(
  evaluation: EvaluationDetailApiResponse
): EvaluationSummary {
  return {
    id: evaluation.id,
    runAt: evaluation.created_at,
    baselineOverall: evaluation.baseline_overall,
    advancedOverall: evaluation.advanced_overall,
    metrics: evaluation.metrics.map(mapEvaluationMetric),
    cases: evaluation.cases.map(mapEvaluationCase),
  };
}

File: backend/app/schemas/evaluation.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class EvaluationMetricResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    evaluation_id: UUID
    key: str
    label: str
    unit: str
    baseline: float
    advanced: float
    higher_is_better: bool
    created_at: datetime


class EvaluationCaseResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    evaluation_id: UUID
    case_id: str
    case_name: str
    description: str
    status: str
    baseline_score: float
    advanced_score: float
    improvement: float
    baseline_tp: int
    baseline_fp: int
    baseline_fn: int
    advanced_tp: int
    advanced_fp: int
    advanced_fn: int
    expected_rules: list | None
    baseline_rules: list | None
    advanced_rules: list | None
    created_at: datetime


class EvaluationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    status: str
    benchmark_version: str
    primary_metric: str
    baseline_overall: float
    advanced_overall: float
    human_time_baseline: float
    human_time_advanced: float
    cost_baseline: float
    cost_advanced: float
    configuration: dict | None
    created_at: datetime


class EvaluationDetailResponse(EvaluationResponse):
    metrics: list[EvaluationMetricResponse]
    cases: list[EvaluationCaseResponse]
